### ISSUE TITLE
Edit content type page crashes while saving...

### ISSUE THREAD

**Issue by drarsalan** [OrchardCMS/OrchardCore#14848]
When I click on a new content type definition that is empty, and click on save, the page crashes.

![image](https://github.com/OrchardCMS/OrchardCore/assets/5498141/53ba052d-7e41-4993-9118-73102b39e652)

```
System.NullReferenceException: Object reference not set to an instance of an object.

   at OrchardCore.ContentTypes.Services.ContentDefinitionService.<>c__DisplayClass28_0.<AlterTypePartsOrderAsync>b__0(ContentTypeDefinitionBuilder type) in E:\Projects\OrchardCore\src\OrchardCore.Modules\OrchardCore.ContentTypes\Services\ContentDefinitionService.cs:line 393

   at OrchardCore.ContentManagement.Metadata.ContentDefinitionManagerExtensions.AlterTypeDefinitionAsync(IContentDefinitionManager manager, String name, Action`1 alteration) in E:\Projects\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement.Abstractions\ContentDefinitionManagerExtensions.cs:line 33

   at OrchardCore.ContentTypes.Controllers.AdminController.EditPOST(String id, EditTypeViewModel viewModel) in E:\Projects\OrchardCore\src\OrchardCore.Modules\OrchardCore.ContentTypes\Controllers\AdminController.cs:line 203

   at lambda_method724(Closure, Object)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()

--- End of stack trace from previous location ---
```

**Comment by hishamco** [comment]
Please steps to reproduce, also the Orchard Core version

**Comment by MikeAlhayek** [comment]
I think this was fixed last night via PR #14837. If you are using 1.8 previews, please test with the latest

**Comment by sebastienros** [comment]
Fixed in 1.8 (main branch)
